packages/@aws-cdk/aws-ecs/test/cluster.test.ts

```diff
@@ -2155,6 +2155,163 @@ describe('cluster', () => {
 
   });
 
+  test('should throw an error if capacity provider with default strategy is not present in capacity providers', () => {
```

Comment:
can we also add a test case and some error validation to catch the case where the customer specifies a mix of Fargate and EC2 ASG capacity providers in the default strategy? Strategies can only contain ASG providers or Fargate + Fargate Spot.